main.c

```diff
@@ -676,15 +744,16 @@ static void kbd_key(void *data, struct wl_keyboard *k, uint32_t serial,
 		return;
 	}
 
-	if (state == WL_KEYBOARD_KEY_STATE_RELEASED) {
-		if (sym == term.repeat.sym)
-			reset_repeat();
-	} else if (state == WL_KEYBOARD_KEY_STATE_PRESSED) {
+	if (state == WL_KEYBOARD_KEY_STATE_PRESSED) {
 		unicode = xkb_keysym_to_utf32(sym);
 
-		term.repeat.sym = sym;
-		term.repeat.unicode = unicode;
-		timerfd_settime(term.repeat.fd, 0, &term.repeat.its, NULL);
+		if (xkb_compose_state_get_status(term.xkb_compose_state) != XKB_COMPOSE_COMPOSING) {
```

Comment:
this can segfault here if xkb_compose_state is NULL, for example when a compose file for the locale is missing.